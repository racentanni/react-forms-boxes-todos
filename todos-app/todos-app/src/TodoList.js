import React, { useState, useEffect } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import './TodoList.scss';

function TodoList() {
  const [todos, setTodos] = useState(() => {
    const savedTodos = localStorage.getItem('todos');
    return savedTodos ? JSON.parse(savedTodos) : [];
  });

  useEffect(() => {
    localStorage.setItem('todos', JSON.stringify(todos));
  }, [todos]);

  const addTodo = (task) => {
    setTodos([...todos, { id: Date.now(), task }]);
  };

  const removeTodo = (id) => {
    setTodos(todos.filter(todo => todo.id !== id));
  };

  const updateTodo = (id, newTask) => {
    setTodos(todos.map(todo => (todo.id === id ? { ...todo, task: newTask } : todo)));
  };

  return (
    <div className='TodoList'>
        <h1>Todo List</h1>
      <NewTodoForm addTodo={addTodo} />
      {todos.map(todo => (
        <Todo
          key={todo.id}
          id={todo.id}
          task={todo.task}
          removeTodo={removeTodo}
          updateTodo={updateTodo}
        />
      ))}
    </div>
  );
}

export default TodoList;