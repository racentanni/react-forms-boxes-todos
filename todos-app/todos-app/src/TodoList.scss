.TodoList {
    max-width: 600px;
    margin: 0 auto;
    padding: 20px;
    background-color: #f9f9f9;
    border: 1px solid #ddd;
    border-radius: 8px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  
    h1 {
      text-align: center;
      font-size: 24px;
      color: #333;
      margin-bottom: 20px;
    }
  
    .Todo {
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 10px;
      margin-bottom: 10px;
      background-color: #fff;
      border: 1px solid #ddd;
      border-radius: 4px;
      transition: background-color 0.3s ease;
  
      &:hover {
        background-color: #f1f1f1;
      }
  
      span {
        flex-grow: 1;
        font-size: 16px;
        color: #333;
      }
  
      button {
        margin-left: 10px;
        padding: 5px 10px;
        font-size: 14px;
        color: #fff;
        background-color: #007bff;
        border: none;
        border-radius: 4px;
        cursor: pointer;
        transition: background-color 0.3s ease;
  
        &:hover {
          background-color: #0056b3;
        }
  
        &.remove {
          background-color: #dc3545;
  
          &:hover {
            background-color: #c82333;
          }
        }
  
        &.edit {
          background-color: #ffc107;
  
          &:hover {
            background-color: #e0a800;
          }
        }
  
        &.complete {
          background-color: #28a745;
  
          &:hover {
            background-color: #218838;
          }
        }
      }
    }
  
    form {
      display: flex;
      justify-content: space-between;
      margin-bottom: 20px;
  
      input {
        flex-grow: 1;
        padding: 10px;
        font-size: 16px;
        border: 1px solid #ddd;
        border-radius: 4px;
        margin-right: 10px;
      }
  
      button {
        padding: 10px 20px;
        font-size: 16px;
        color: #fff;
        background-color: #007bff;
        border: none;
        border-radius: 4px;
        cursor: pointer;
        transition: background-color 0.3s ease;
  
        &:hover {
          background-color: #0056b3;
        }
      }
    }
  }