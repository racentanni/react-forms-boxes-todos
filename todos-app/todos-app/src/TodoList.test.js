import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

test('renders without crashing', () => {
  render(<TodoList />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test('can add a new todo', () => {
  const { getByPlaceholderText, getByText } = render(<TodoList />);
  const input = getByPlaceholderText('New Todo');
  const button = getByText('Add Todo');

  fireEvent.change(input, { target: { value: 'Test Todo' } });
  fireEvent.click(button);

  expect(getByText('Test Todo')).toBeInTheDocument();
});

test('can remove a todo', () => {
  const { getByPlaceholderText, getByText, queryByText, getByLabelText } = render(<TodoList />);
  const input = getByPlaceholderText('New Todo');
  const button = getByText('Add Todo');

  fireEvent.change(input, { target: { value: 'Test Todo' } });
  fireEvent.click(button);

  // Assuming each todo item has a label with the text of the todo
  const removeButton = getByLabelText('Remove Test Todo');
  fireEvent.click(removeButton);

  expect(queryByText('Test Todo')).not.toBeInTheDocument();
});