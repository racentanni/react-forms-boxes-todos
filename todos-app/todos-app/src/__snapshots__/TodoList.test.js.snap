// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1>
      Todo List
    </h1>
    <form>
      <input
        placeholder="New Todo"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
